cupy/core/include/cupy/type_dispatcher.cuh

```diff
@@ -40,11 +40,11 @@ void dtype_dispatcher(int dtype_id, functor_t f, Ts&&... args)
     case CUPY_TYPE_INT8:       return f.template operator()<char>(std::forward<Ts>(args)...);
     case CUPY_TYPE_INT16:      return f.template operator()<short>(std::forward<Ts>(args)...);
     case CUPY_TYPE_INT32:      return f.template operator()<int>(std::forward<Ts>(args)...);
-    case CUPY_TYPE_INT64:      return f.template operator()<long>(std::forward<Ts>(args)...);
+    case CUPY_TYPE_INT64:      return f.template operator()<int64_t>(std::forward<Ts>(args)...);
```

Comment:
The header <cstdint> will be required.

Reply:
Yeah I was puzzled about this...All checks have passed, do you know why?

Reply:
I noticed both `cupy/cuda/thrust.cpp` and `cupy/cuda/cub.cpp` include `stdint.h` for some reason, perhaps exposed through other `cimport`?

Reply:
Done in d760b14.